src/cmd/services/m3dbnode/config/config.go

```diff
@@ -387,6 +387,14 @@ type IndexConfiguration struct {
 	// as they are very CPU-intensive (regex and FST matching).
 	MaxQueryIDsConcurrency int `yaml:"maxQueryIDsConcurrency" validate:"min=0"`
 
+	// MaxResultsPerPermit is the maximum index results a query can process after obtaining a permit. If a query needs
```

Comment:
Should we split this up into different values for 1) regular queries 2) aggregate queries?

Aggregate queries when scoped with `match[]=...` are relatively more expensive per iteration since they have to do postings list interception with each aggregate term which is progressed to, as per:
https://github.com/m3db/m3/blob/812d585fddc53d7a533d69bde777efbfbc7bf29e/src/dbnode/storage/index/fields_terms_iterator.go#L229-L248 (term progression)
and
https://github.com/m3db/m3/blob/812d585fddc53d7a533d69bde777efbfbc7bf29e/src/dbnode/storage/index/fields_terms_iterator.go#L149-L177 (field progression)

Reply:
I suppose we could do this in a followup? Might be too much to add to scope of this PR.

Reply:
that's pretty easy to do now, just 2 different config values.